### ISSUE TITLE
Consider removing henrikbjorn/lurker

### ISSUE THREAD

**Issue by greg-1-anderson** [Codegyre/Robo#272]
Robo uses `require-dev` and `suggest` rather than `require` for most dependent libraries that are not used by the Robo runtime.  henrikbjorn/lurker is the one exception to this; it is brought in via `require`, even though it is largely unused in Robo core.

At a minimum, this project requirements should be moved to `require-dev`, and added to `suggest` in composer.json. This might also imply relocation of Watch.php out of Base.

I would also suggest that lurker is out-of-place as a task.  It seems that there is only nominal benefit to deferring the call to `listener()` to the `run()` method, and overall, the Watcher API is not much more convenient than using `ResourceWatcher` directly.  I think it would be fine to just have documentation on how to use ResourceWatcher, and `suggest` but not use lurker in Robo core.  

If a ResourceWatcher wrapper is desired, maybe it should not be a task; maybe it should be a non-task wrapper that takes a TaskInterface as its parameter.  In this way, you could define a Collection of tasks to execute, and pass this collection to the ResourceWatcher wrapper.

**Comment by burzum** [comment]
If you remove it it must be ensured that watchers will still work out of the box - IMHO.

**Comment by greg-1-anderson** [comment]
[ResourceWatcher](https://github.com/henrikbjorn/Lurker/blob/master/src/Lurker/ResourceWatcher.php) is the class provided by henrikbjorn/lurker.

If we remove henrikbjorn/lurker from the `require` section of Robo's composer.json, then any project currently using it would have to add it to its own composer.json project.

If we removed or rewrote the wrapper Robo provides for henrikbjorn/lurker, then any Robo project that used it would have to be adjusted to use the new mechanism, or use henrikbjorn/lurker directly. To make it easier to adapt existing code, we could consider providing a robo-lurker project containing the old API.  If we did this, then existing code would only need to a) add the new project to their composer.json project, and b) `use` the resource watcher wrapper class in their RoboFile.php.

Removing henrikbjorn/lurker would promote better loose coupling between components.  This is currently the only project that Robo requires, but does not use; this is why I recommend removing it prior to making a stable Robo release. If removal is desired, I'll submit a PR to remove it that introduces an "Upgrading" section in the documentation (referenced from the README) that can be enhanced every time Robo introduces a breaking change.
